packages/dev/core/src/PostProcesses/screenSpaceReflectionPostProcess.ts

```diff
@@ -11,40 +11,153 @@ import { ScreenSpaceReflectionsConfiguration } from "../Rendering/screenSpaceRef
 
 import "../Shaders/screenSpaceReflection.fragment";
 import { RegisterClass } from "../Misc/typeStore";
+import type { CubeTexture } from "../Materials/Textures/cubeTexture";
 
 declare type Engine = import("../Engines/engine").Engine;
 declare type Scene = import("../scene").Scene;
 
 /**
- * The ScreenSpaceReflectionPostProcess performs realtime reflections using only and only the available informations on the screen (positions and normals).
- * Basically, the screen space reflection post-process will compute reflections according the material's reflectivity.
+ * The ScreenSpaceReflectionPostProcess performs realtime reflections using only the available informations on the screen (positions, depth and normals).
+ * Basically, the screen space reflection post-process will compute reflections according the material's properties.
  */
 export class ScreenSpaceReflectionPostProcess extends PostProcess {
+    @serialize()
+    private _backwardCompatibility: Nullable<Boolean> = true;
+
+    /**
+     * Sets the boolean deciding if we should use the old SSR version or the new one. Default value is true.
+     */
+    public set backwardCompatibility(backCompat: Nullable<boolean>) {
+        this._backwardCompatibility = backCompat;
+        this._updateEffectDefines();
+    }
     /**
      * Gets or sets a reflection threshold mainly used to adjust the reflection's height.
+     * For backward compatibility only. Default value is 1.2.
      */
     @serialize()
     public threshold: number = 1.2;
+
     /**
-     * Gets or sets the current reflection strength. 1.0 is an ideal value but can be increased/decreased for particular results.
+     * Gets or sets the step size used to iterate until the effect finds the color of the reflection's pixel. Typically in interval [0.1, 1.0]. Default value is 1.0.
+     * For backward compatibility only.
      */
     @serialize()
-    public strength: number = 1;
+    public step: number = 1.0;
+
+    private _enableSmoothReflections: boolean = false;
+    private _reflectionSamples: number = 64;

```

Comment:
"reflectionSamples" is missing getter and setter. We need it for backward compatibility